### PR TITLE
Use Strimzi Test Container in TO tests

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Task

### Description

We should use Strimzi Test containers in our tests that need to use Kafka. This will give us two main benefits:
* We will be dogfooding our own tools and that will help us to improve them (a bunch of improvements was already raised from work on this PR)
* We will get proper isolation between the Kafka version used in tests and Kafka version used by the operator. Thanks to that, the unit/integration tests will not be causing problems and blocking us with every Kafka release.

### Checklist

- [x] Make sure all tests pass

### CODE REVIEW

File: .azure/templates/jobs/build/test_strimzi.yaml [hunk 1]
```diff
@@ -29,7 +29,7 @@ jobs:
       - bash: "mvn -e -V -B -Dmaven.javadoc.skip=true -Dsurefire.rerunFailingTestsCount=5 -Dfailsafe.rerunFailingTestsCount=2 install"
         env:
           # Test container optimization and also we eliminate pulling RYUK image from DockerHub to prevent limits
-          TESTCONTAINERS_RYUK_DISABLED: TRUE
+          TESTCONTAINERS_RYUK_DISABLED: FALSE
```
Comment (see-quick) [inline .azure/templates/jobs/build/test_strimzi.yaml:32]: Why is RYUK now enabled? :).

Comment (scholzj) [reply .azure/templates/jobs/build/test_strimzi.yaml:32]: RYUK is doing some cleanup of the containers and networks. I originally enabled it before we had the shared network support. Let me see if it is still needed or not.

Comment (see-quick) [reply .azure/templates/jobs/build/test_strimzi.yaml]: Sure, thanks.

Comment (scholzj) [reply .azure/templates/jobs/build/test_strimzi.yaml]: @see-quick Looks like it works, so let's keep RYUK disabled.

**Review by see-quick — APPROVED** [review]

**Pull request merged**